include/wally_descriptor.h

```diff
@@ -0,0 +1,172 @@
+ * :param output: Destination for the resulting parsed descriptor.
+ *|    The descriptor returned should be freed using `wally_descriptor_free`.
+ *
+ * Variable names can be used in the descriptor string and will be substituted
+ * with the contents of ``vars_in`` during parsing. Key names for ``vars_in``
+ * must be 16 characters or less in length and start with a letter.
+ */
+WALLY_CORE_API int wally_descriptor_parse(
+    const char *descriptor,
+    const struct wally_map *vars_in,
+    uint32_t network,
+    uint32_t flags,
+    struct wally_descriptor **output);
+
+/**
+ * Free a parsed output descriptor or miniscript expression.
+ *
+ * :param descriptor: Parsed output descriptor or miniscript expression to free.
+ */
+WALLY_CORE_API int wally_descriptor_free(
+    struct wally_descriptor *descriptor);
+
+/**
+ * Canonicalize a descriptor.
+ *
+ * :param descriptor: Parsed output descriptor or miniscript expression.
+ * :param flags: For future use. Must be 0.
+ * :param output: Destination for the resulting canonical descriptor.
+ *|    The string returned should be freed using `wally_free_string`.
+ *
+ * .. note:: Other canonicalization (hardened derivation indicator
+ * mapping, and private to public key mapping) is not yet implemented.
+ */
+WALLY_CORE_API int wally_descriptor_canonicalize(
+    const struct wally_descriptor *descriptor,
+    uint32_t flags,
+    char **output);
+
+/**
+ * Create an output descriptor checksum.
+ *
+ * :param descriptor: Parsed output descriptor or miniscript expression.
+ * :param flags: For future use. Must be 0.
+ * :param output: Destination for the resulting descriptor checksum.
+ *|    The string returned should be freed using `wally_free_string`.
+ */
+WALLY_CORE_API int wally_descriptor_get_checksum(
+    const struct wally_descriptor *descriptor,
+    uint32_t flags,
+    char **output);
+
+/**
+ * Get the features used in a parsed output descriptor or miniscript expression.
+ *
+ * :param descriptor: Parsed output descriptor or miniscript expression.
+ * :param value_out: Destination for the resulting ``WALLY_MS_`` feature flags.
+ */
+WALLY_CORE_API int wally_descriptor_get_features(
+    const struct wally_descriptor *descriptor,
+    uint32_t *value_out);
+
+/**
+ * Get the maximum length of a script corresponding to an output descriptor.
+ *
+ * :param descriptor: Parsed output descriptor or miniscript expression.
+ * :param flags: For future use. Must be 0.
+ * :param written: Destination for the resulting maximum script length.
+ *
+ * .. note:: This function overestimates the script size required, but is
+ *|    cheap to compute (does not require script generation).
+ */
+WALLY_CORE_API int wally_descriptor_to_script_get_maximum_length(
+    const struct wally_descriptor *descriptor,
+    uint32_t flags,
+    size_t *written);
+
+/**
+ * Create a script corresponding to an output descriptor or miniscript expression.
```

Comment:
Which script is created? 

rust-miniscript allows to generate scriptpubkey, (unsigned) script sig, explicit script/witness script/redeem script, script code, etc. I guess here it's the scriptpubkey, if so I think it's worth mentioning it in the docs and perhaps in the function(s) name

Reply:
So IIUC you can control the ("sub")script generated with `depth` and `index`. It wasn't very clear to me without looking at the tests, perhaps it's worth expanding a bit the documentation here.

Reply:
Added documentation notes for both. Generating other script types in the future will be handled via the `flags` argument rather than separate functions for each type.